sdk/resourcemanager/Azure.ResourceManager.Core/tests/Scenario/GenericResourceTests.cs

```diff
@@ -0,0 +1,88 @@
+using System.Runtime.Versioning;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Identity;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests
+{
+    public class GenericResourceTests : ResourceManagerTestBase
+    {
+        private string _rgName;
+
+        private readonly string _location = "southcentralus";
+
+        public GenericResourceTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [OneTimeSetUp]
+        public void LocalOneTimeSetup()
+        {
+            _rgName = SessionRecording.GenerateAssetName("testRg-");
+            var subscriptionOperations = GlobalClient.GetSubscriptionOperations(SessionEnvironment.SubscriptionId);
+            _ = subscriptionOperations.GetResourceGroupContainer().Construct(_location).StartCreateOrUpdateAsync(_rgName).ConfigureAwait(false).GetAwaiter().GetResult().Value;
+            StopSessionRecording();
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public void GetGenerics()
+        {
+            AzureResourceManagerClientOptions options = new AzureResourceManagerClientOptions();
+            _ = GetArmClient(options); // setup providers client
+            var asetid = $"/subscriptions/{TestEnvironment.SubscriptionId}/resourceGroups/{_rgName}/providers/Microsoft.Compute/availabilitySets/testavset";
+            var subOp = Client.GetSubscriptionOperations(TestEnvironment.SubscriptionId);
+            var genericResourceOperations = new GenericResourceOperations(subOp, asetid);
+            Assert.ThrowsAsync<RequestFailedException>(async () => await genericResourceOperations.GetAsync());
```

Comment:
assert its a 404 to filter out all other ways it can fail

Reply:
Just want to test an exception is thrown here, next test does the 404 check